### PR TITLE
zoom-us: fix launch

### LINKED ISSUE #79954: Zoom won't start due to LD_PRELOAD of v4l2convert
**Describe the bug**
i'm trying to start zoom and it does not start. After looking into the wrapper script and messing with it I was able to get it to open and work fine.

**To Reproduce**
1. Try to run `/nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/ZoomLauncher` built from nixpkgs-unstable@cc1ae9f21b9e0ce998e706a3de1bad0b5259f22d

**Expected behavior**
Zoom starts up.

**Screenshots**
```
[~]─> sha1sum ZoomLauncher /nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/ZoomLauncher
d2ec8af7569d94eca0b0fd10a5c7625e0ea4212e  ZoomLauncher
d2ec8af7569d94eca0b0fd10a5c7625e0ea4212e  /nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/ZoomLauncher
[~]─> cat ZoomLauncher 
#! /nix/store/r6z3c7dwdxw4jjdg1l8nqwxm83j4gll8-bash-4.4-p23/bin/bash -e
export PATH='/nix/store/pabqc2cx3jrqijii1il528pd2by8ppg5-coreutils-8.31/bin:/nix/store/gifkpxcc4jfqr3fkb1s6xhnwy9nz3k2c-glib-2.62.4-dev/bin:/nix/store/sjpi0lagbh7qrs1jviwfi7nlw1bvz8gl-pciutils-3.6.4/bin:/nix/store/x39gmvjnijsb9rk2m5a7mgfcshbjxgxj-procps-3.3.16/bin:/nix/store/ghc8hp7q3fshs5m2zzn9ygd06flrpf1x-qttools-5.12.6-dev/bin:/nix/store/jv3my2ca8s3wch314cv09spc2sdablig-util-linux-2.33.2-bin/bin'${PATH:+':'}$PATH
export LD_PRELOAD='/nix/store/nqk9fi13abpa6pq9mqypvmpv5ibq6x7d-v4l-utils-1.18.0/lib/libv4l/v4l2convert.so'${LD_PRELOAD:+':'}$LD_PRELOAD
export QT_PLUGIN_PATH='/nix/store/7xbn9ly6f6wqf3mxx1g6ddryd8vg1i0c-qtbase-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/0z4ylv2n9vnh0hwjxdd9j7i6wysclnk0-qtsvg-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QT_PLUGIN_PATH='/nix/store/qxk0xmg9i54s5wwayjciya3g304xc5g8-qtdeclarative-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/qxk0xmg9i54s5wwayjciya3g304xc5g8-qtdeclarative-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/7wripwhy4rh5qhqhxmc36zk7kwp4nzkg-qtquickcontrols-5.12.6/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/jy5x5xaxl7v1m89scnz5ngqsphpabald-qtwayland-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/jy5x5xaxl7v1m89scnz5ngqsphpabald-qtwayland-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/zc2dxqlwmdgs0pqiikqmkfxpwxj3hpcq-qtmultimedia-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/zc2dxqlwmdgs0pqiikqmkfxpwxj3hpcq-qtmultimedia-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/bs17yps0pm4bygmwinp57ni2jr482wff-qtlocation-5.12.6-bin/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH='/nix/store/bs17yps0pm4bygmwinp57ni2jr482wff-qtlocation-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/b6qi5vchzl0fkg401yhai67r14kr323m-qtquickcontrols2-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/pv247k834y0f8ahfsm96506072ji3wip-qtwebchannel-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QML2_IMPORT_PATH='/nix/store/lsl6scfznk1i73a91svgs8zpk62jxszd-qtwebengine-5.12.6-bin/lib/qt-5.12.6/qml'${QML2_IMPORT_PATH:+':'}$QML2_IMPORT_PATH
export QT_PLUGIN_PATH='/nix/store/r2mhb5hhfsmpaih3hjvyzn89zkpsvpp7-qtimageformats-5.12.6/lib/qt-5.12.6/plugins'${QT_PLUGIN_PATH:+':'}$QT_PLUGIN_PATH
export XDG_DATA_DIRS='/nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
cd /nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us
exec -a "$0" "/nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/.ZoomLauncher-wrapped"  "$@"
[~]─> ./ZoomLauncher 
ZoomLauncher started.
cmd line: 
CreateReportChannel bp_server_fd=4
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
$HOME =  
Can't load/.config/zoomus.conf
export SSB_HOME=/.zoom; export QSG_INFO=1; export LD_LIBRARY_PATH=/nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us; export BREAKPAD_CLIENT_FD=3; /nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/zoom "" 
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
success to create child process,status is 32512.
zoom exited normally.
Something went wrong while running zoom,exit code is 127.
ZoomLauncher exit.
[~]─> sed -i '/LD_PRELOAD/ s|^|#|' ZoomLauncher 
[~]─> ./ZoomLauncher 
ZoomLauncher started.
cmd line: 
CreateReportChannel bp_server_fd=4
$HOME = /home/manny 
export SSB_HOME=/home/manny/.zoom; export QSG_INFO=1; export LD_LIBRARY_PATH=/nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us; export BREAKPAD_CLIENT_FD=3; /nix/store/k5nvbc2p833q79g344cj98s45v8mrvhy-zoom-us-3.5.352596.0119/share/zoom-us/zoom "" 
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
zoom started.
Client: Breakpad is using Client-Server Mode! client fd = 3
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
[CZPClientLogMgr::LogClientEnvironment] [MacAddr: 9C:B6:D0:97:A3:3D][client: Linux][OS: NixOS 20.03 (Markhor)][Hardware: CPU Core:4 Frenquency:1.8 G Memory size:15705MB CPU Brand:Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz GPU Brand:][Req ID: ]
Linux Client Version is 3.5.352596.0119
QSG_RENDER_LOOP is 
XDG_CURRENT_DESKTOP = ;   GDMSESSION = 
Graphics Card Info:: 00:02.0 VGA compatible controller: Intel Corporation UHD Graphics 620 (Whiskey Lake)
Zoom package arch is 64bit, runing OS arch is x86_64
AppIconMgr::systemDesktopName log Desktop Name:  
sh: relocation error: /nix/store/px0i6m2d92325zigdfa2r9jkdmws30my-glibc-2.30/lib/libpthread.so.0: symbol __write_nocancel version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
qt.scenegraph.general: threaded render loop
qt.scenegraph.general: Using sg animation driver
qt.scenegraph.general: Animation Driver: using vsync: 16.95 ms
qt.scenegraph.general: Using sg animation driver
qt.scenegraph.general: Animation Driver: using vsync: 16.95 ms
qt.scenegraph.general: texture atlas dimensions: 1024x512
qt.scenegraph.general: R/G/B/A Buffers:   8 8 8 8
qt.scenegraph.general: Depth Buffer:      24
qt.scenegraph.general: Stencil Buffer:    8
qt.scenegraph.general: Samples:           -1
qt.scenegraph.general: GL_VENDOR:         Intel Open Source Technology Center
qt.scenegraph.general: GL_RENDERER:       Mesa DRI Intel(R) UHD Graphics (Whiskey Lake 3x8 GT2) 
qt.scenegraph.general: GL_VERSION:        3.0 Mesa 19.3.3
qt.scenegraph.general: GL_EXTENSIONS:    GL_3DFX_texture_compression_FXT1 GL_AMD_conservative_depth GL_AMD_depth_clamp_separate GL_AMD_draw_buffers_blend GL_AMD_multi_draw_indirect GL_AMD_query_buffer_object GL_AMD_seamless_cubemap_per_texture GL_AMD_shader_stencil_export GL_AMD_shader_trinary_minmax GL_AMD_texture_texture4 GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_APPLE_object_purgeable GL_APPLE_packed_pixels GL_ARB_ES2_compatibility GL_ARB_ES3_1_compatibility GL_ARB_ES3_compatibility GL_ARB_arrays_of_arrays GL_ARB_blend_func_extended GL_ARB_buffer_storage GL_ARB_clear_buffer_object GL_ARB_clear_texture GL_ARB_clip_control GL_ARB_color_buffer_float GL_ARB_compressed_texture_pixel_storage GL_ARB_compute_shader GL_ARB_conditional_render_inverted GL_ARB_conservative_depth GL_ARB_copy_buffer GL_ARB_copy_image GL_ARB_cull_distance GL_ARB_debug_output GL_ARB_depth_buffer_float GL_ARB_depth_clamp GL_ARB_depth_texture GL_ARB_derivative_control GL_ARB_draw_buffers GL_ARB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_draw_indirect GL_ARB_draw_instanced GL_ARB_explicit_attrib_location GL_ARB_explicit_uniform_location GL_ARB_fragment_coord_conventions GL_ARB_fragment_layer_viewport GL_ARB_fragment_program GL_ARB_fragment_program_shadow GL_ARB_fragment_shader GL_ARB_fragment_shader_interlock GL_ARB_framebuffer_no_attachments GL_ARB_framebuffer_object GL_ARB_framebuffer_sRGB GL_ARB_get_program_binary GL_ARB_get_texture_sub_image GL_ARB_half_float_pixel GL_ARB_half_float_vertex GL_ARB_indirect_parameters GL_ARB_instanced_arrays GL_ARB_internalformat_query GL_ARB_internalformat_query2 GL_ARB_invalidate_subdata GL_ARB_map_buffer_alignment GL_ARB_map_buffer_range GL_ARB_multi_bind GL_ARB_multi_draw_indirect GL_ARB_multisample GL_ARB_multitexture GL_ARB_occlusion_query GL_ARB_occlusion_query2 GL_ARB_parallel_shader_compile GL_ARB_pipeline_statistics_query GL_ARB_pixel_buffer_object GL_ARB_point_parameters GL_ARB_point_sprite GL_ARB_polygon_offset_clamp GL_ARB_post_depth_coverage GL_ARB_program_interface_query GL_ARB_provoking_vertex GL_ARB_query_buffer_object GL_ARB_robust_buffer_access_behavior GL_ARB_robustness GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_seamless_cubemap_per_texture GL_ARB_separate_shader_objects GL_ARB_shader_atomic_counter_ops GL_ARB_shader_atomic_counters GL_ARB_shader_ballot GL_ARB_shader_bit_encoding GL_ARB_shader_clock GL_ARB_shader_draw_parameters GL_ARB_shader_group_vote GL_ARB_shader_image_load_store GL_ARB_shader_image_size GL_ARB_shader_objects GL_ARB_shader_precision GL_ARB_shader_stencil_export GL_ARB_shader_storage_buffer_object GL_ARB_shader_texture_image_samples GL_ARB_shader_texture_lod GL_ARB_shading_language_100 GL_ARB_shading_language_420pack GL_ARB_shading_language_packing GL_ARB_shadow GL_ARB_stencil_texturing GL_ARB_sync GL_ARB_texture_barrier GL_ARB_texture_border_clamp GL_ARB_texture_compression GL_ARB_texture_compression_bptc GL_ARB_texture_compression_rgtc GL_ARB_texture_cube_map GL_ARB_texture_cube_map_array GL_ARB_texture_env_add GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_filter_anisotropic GL_ARB_texture_float GL_ARB_texture_gather GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_mirrored_repeat GL_ARB_texture_multisample GL_ARB_texture_non_power_of_two GL_ARB_texture_query_levels GL_ARB_texture_query_lod GL_ARB_texture_rectangle GL_ARB_texture_rg GL_ARB_texture_rgb10_a2ui GL_ARB_texture_stencil8 GL_ARB_texture_storage GL_ARB_texture_storage_multisample GL_ARB_texture_swizzle GL_ARB_texture_view GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_transform_feedback_instanced GL_ARB_transform_feedback_overflow_query GL_ARB_transpose_matrix GL_ARB_uniform_buffer_object GL_ARB_vertex_array_bgra GL_ARB_vertex_array_object GL_ARB_vertex_attrib_binding GL_ARB_vertex_buffer_object GL_ARB_vertex_program GL_ARB_vertex_shader GL_ARB_vertex_type_10f_11f_11f_rev GL_ARB_vertex_type_2_10_10_10_rev GL_ARB_window_pos GL_ATI_blend_equation_separate GL_ATI_draw_buffers GL_ATI_separate_stencil GL_ATI_texture_env_combine3 GL_ATI_texture_float GL_EXT_EGL_sync GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_equation_separate GL_EXT_blend_func_separate GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_compiled_vertex_array GL_EXT_copy_texture GL_EXT_demote_to_helper_invocation GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_draw_range_elements GL_EXT_fog_coord GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_framebuffer_object GL_EXT_framebuffer_sRGB GL_EXT_gpu_program_parameters GL_EXT_multi_draw_arrays GL_EXT_packed_depth_stencil GL_EXT_packed_float GL_EXT_packed_pixels GL_EXT_pixel_buffer_object GL_EXT_point_parameters GL_EXT_polygon_offset_clamp GL_EXT_provoking_vertex GL_EXT_rescale_normal GL_EXT_secondary_color GL_EXT_separate_specular_color GL_EXT_shader_framebuffer_fetch GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_shader_integer_mix GL_EXT_shader_samples_identical GL_EXT_shadow_funcs GL_EXT_stencil_two_side GL_EXT_stencil_wrap GL_EXT_subtexture GL_EXT_texture GL_EXT_texture3D GL_EXT_texture_array GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_compression_s3tc GL_EXT_texture_cube_map GL_EXT_texture_edge_clamp GL_EXT_texture_env_add GL_EXT_texture_env_combine GL_EXT_texture_env_dot3 GL_EXT_texture_filter_anisotropic GL_EXT_texture_integer GL_EXT_texture_lod_bias GL_EXT_texture_object GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_decode GL_EXT_texture_shadow_lod GL_EXT_texture_shared_exponent GL_EXT_texture_snorm GL_EXT_texture_swizzle GL_EXT_timer_query GL_EXT_transform_feedback GL_EXT_vertex_array GL_EXT_vertex_array_bgra GL_IBM_multimode_draw_arrays GL_IBM_rasterpos_clip GL_IBM_texture_mirrored_repeat GL_INGR_blend_func_separate GL_INTEL_performance_query GL_INTEL_shader_atomic_float_minmax GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_debug GL_KHR_no_error GL_KHR_parallel_shader_compile GL_KHR_robust_buffer_access_behavior GL_KHR_robustness GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_sliced_3d GL_MESA_pack_invert GL_MESA_shader_integer_functions GL_MESA_texture_signed_rgba GL_MESA_window_pos GL_NV_blend_square GL_NV_compute_shader_derivatives GL_NV_conditional_render GL_NV_depth_clamp GL_NV_fog_distance GL_NV_fragment_shader_interlock GL_NV_light_max_exponent GL_NV_packed_depth_stencil GL_NV_primitive_restart GL_NV_texgen_reflection GL_NV_texture_barrier GL_NV_texture_env_combine4 GL_NV_texture_rectangle GL_OES_EGL_image GL_OES_read_format GL_S3_s3tc GL_SGIS_generate_mipmap GL_SGIS_texture_border_clamp GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_SUN_multi_draw_arrays
qt.scenegraph.general: Max Texture Size: 16384
qt.scenegraph.general: Debug context:    false
Error: Send error, 22 Invalid argument
Error: Send error, 22 Invalid argument
Error: Send error, 22 Invalid argument
^Csuccess to create child process,status is 2.
zoom  exited abnormally.
ZoomLauncher exit.
Exception: interrupted
[tty], line 1: ./ZoomLauncher 
[~]─> 
```

**Additional context**
With LD_PRELOAD commented out, the SSB_HOME is corectly set to $HOME/..zoom where as otherwise it is being set incorrectly.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
[~]─> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.2, NixOS, 20.03pre212208.8130f3c1c2b (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-20.03pre212208.8130f3c1c2b"`
 - channels(manny): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - @danbst 
  - @tadfisher 
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by danbst**
Probably due to glibc update, ZoomLauncher became broken when v4l is present in
LD_PRELOAD path. It can be fixed by a) removing ZoomLauncher from startup chain,
so `zoom` is started directly or b) removing v4l from LD_PRELOAD.

The reason v4l was added before was because my video was rotated upside down without it.
Seem like nowadays this is fixed by Zoom itself, so I'm disabling it.

Fixes https://github.com/NixOS/nixpkgs/issues/79954

Co-authored-by: @mmlb

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthewbauer** [comment]
It might make sense to just remove the LD_PRELOAD line. This was needed to support some webcams which aren't the proper pixel format expected in libv4l1 apps. The one I hit this with was the logitech C310. Those might be fixed in zoom-us anyway.

See https://wiki.archlinux.org/index.php/Webcam_setup#V4L1_support

**Comment by mmlb** [comment]
tested and works for me.

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 1]
```diff
@@ -101,11 +102,10 @@ in mkDerivation {
 
   qtWrapperArgs = [
     ''--prefix PATH : ${makeBinPath [ coreutils glib.dev pciutils procps qttools.dev utillinux ]}''
-    ''--prefix LD_PRELOAD : ${libv4l}/lib/libv4l/v4l2convert.so''
     # --run "cd ${placeholder "out"}/share/zoom-us"
     # ^^ unfortunately, breaks run arg into multiple array elements, due to
     # some bad array propagation. We'll do that in bash below
-  ];
+  ] ++ stdenv.lib.optional enableV4l ''--prefix LD_PRELOAD : ${libv4l}/lib/libv4l/v4l2convert.so'';
```
Comment (jtojnar) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Would adding it to `autoPatchElf`’s `runtimeDeps` work. `LD_PRELOAD` is unwanted precisely for the reason described.

Comment (jtojnar) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Though https://github.com/NixOS/nixpkgs/pull/80070 will be needed for this to work.

Comment (jtojnar) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Oh, it is `LD_PRELOAD`ed. Disregard this comment.

**Comment by acowley** [comment]
I needed to re-add this `LD_PRELOAD` to get a Logitech C270 webcam working with zoom-us version 3.5.361976.0301 from nixpkgs ddf87fb1baf8f5022281dad13fb318fa5c17a7c6 (March 18, 2020). Without it, the video preview in zoom-us remains black.

**Comment by matthewbauer** [comment]
> I needed to re-add this `LD_PRELOAD` to get a Logitech C270 webcam working with zoom-us version 3.5.361976.0301 from nixpkgs [ddf87fb](https://github.com/NixOS/nixpkgs/commit/ddf87fb1baf8f5022281dad13fb318fa5c17a7c6) (March 18, 2020). Without it, the video preview in zoom-us remains black.

So that means you aren't getting the error this PR fixes?

**Comment by acowley** [comment]
I’m not sure what the error was from reading this PR. I plugged in the camera, ran `nix-shell -p zoom-us`, and then `zoom-us` in that shell. The program ran, but video remained black. I tried other programs like VLC, and the camera worked there.

I googled a bit, but found nothing. Searched this repo, found this PR, and tried setting `LD_PRELOAD` in the same `nix-shell`. The video now worked. I then added for myself a derivation that just creates a wrapper around the nixpkgs zoom-us, so now I can use that.

My video was initially rotated in zoom, but not in other applications. Zoom has a setting to rotate the video, so I used that.

I’m not sure what the right course of action is as `LD_PRELOAD` is a pretty unsightly fix, but I’m very glad I found this PR as figuring it out on my own would have taken ages.

**Comment by neilmayhew** [comment]
Like @acowley, I have a Logitech C270 and this commit breaks Zoom for me. An overlay restoring the `LD_PRELOAD` of `v4l` fixes it:

```nix
self: super:
{
  zoom-us = super.zoom-us.overrideAttrs (old: rec {
    qtWrapperArgs = ["--prefix LD_PRELOAD : ${self.libv4l}/lib/libv4l/v4l2convert.so"]
      ++ old.qtWrapperArgs;
  });
}
```

Like Anthony says, this is an ugly solution, and I'd like not to have to keep using it for evermore. Is there a better solution, one that will make Zoom work for everybody?

Full disclosure: I'm using `nixos-20.03-small` as the channel for my OS and `nixos-unstable-small` for my personal profile.

**Comment by neilmayhew** [comment]
The problem this PR aims to fix (#79954) seems to have been a result of mixing 19.03 and unstable, as discussed in #84043. However, the problem has gone away with 20.03, and probably never was a direct result of using `v4l`.

**Comment by matthewbauer** [comment]
> The problem this PR aims to fix (#79954) seems to have been a result of mixing 19.03 and unstable, as discussed in #84043. However, the problem has gone away with 20.03, and probably never was a direct result of using `v4l`.

If we're pretty sure this is the case, I think we can re-revert this. Probably should be cautious though, and verify this works in a number of different environments. I am still surprised that the only solution requires LD_PRELOAD! I would expect v4l to have some configuration setting to load the compat mode.

**Pull request merged**